### PR TITLE
Add spring/compress module

### LINKED ISSUE #2770: new module: spring
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by xec-cm**
## PR checklist

Closes #2770 

- [x] This comment contains a description of changes (with reason).
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/spring/compress/main.nf
Comment (nvnieuwk) [inline modules/nf-core/spring/compress/main.nf:41]:
```suggestion
    def input = meta.single_end ? "-i ${fastq1}" : "-i ${fastq1} ${fastq2}"

    """
    spring \\
        -c \\
        -g \\
        -t ${task.cpus} \\
        $args \\
        ${input} \\
        -o ${prefix}.spring

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        spring: ${VERSION}
    END_VERSIONS
    """
```

It's preffered to not have multiple scripts inside one process (especially if it's such a small difference)

File: modules/nf-core/spring/compress/main.nf
Comment (nvnieuwk) [inline modules/nf-core/spring/compress/main.nf:11]:
```suggestion
    tuple val(meta), path(fastq1), path(fastq2)
```

This is a bit more clear

Comment (nvnieuwk) [reply modules/nf-core/spring/compress/main.nf:11]: Can you also update the `meta.yml` for this?

File: tests/modules/nf-core/spring/compress/main.nf
Comment (nvnieuwk) [inline tests/modules/nf-core/spring/compress/main.nf:21]:
```suggestion
        file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true), 
        file(params.test_data['sarscov2']['illumina']['test_2_fastq_gz'], checkIfExists: true)
```

File: tests/modules/nf-core/spring/compress/main.nf
Comment (nvnieuwk) [inline tests/modules/nf-core/spring/compress/main.nf:11]:
```suggestion
        file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true),
        []
```

**Review by nvnieuwk — CHANGES_REQUESTED** [review]

**Comment by xec-cm** [comment]
Dear @nvnieuwk, 

Thanks for reviewing this PR! I really appreciate it. I just wanted to check, would it be alright if I created a new PR for the decompression feature and added it as a subfolder within the "spring" folder? Also, just to clarify, when I tried to open the linked issue, I couldn't add myself as an assignee, is that normal?

Thanks again

Best

**Comment by nvnieuwk** [comment]
Yeah go ahead! To assign yourself to an issue you need to be a member of nf-core. You can ask to be added on Slack in the github-invitations channel 🙂

**Review by nvnieuwk — APPROVED** [review]
LGTM!

**Comment by SPPearce** [comment]
Hi @xec-cm , 
It looks like this PR is quite close to being ready to merge, with just some additional lint changes since you put it together. Let me know if you need any help in finishing it off.

**Comment by xec-cm** [comment]
@SPPearce done!

**Review by SPPearce — APPROVED** [review]
Looks good to me then!

**Comment by SPPearce** [comment]
Can you press the Merge button, or do I need to do it for you?

**Comment by xec-cm** [comment]
Hi @SPPearce

I do not have write access to this repository and therefore cannot merge the PR. Can you please do it?

**Pull request merged**
